langroid/agent/base.py

```diff
@@ -649,23 +657,16 @@ def user_response(
         interactive = self.interactive or settings.interactive
```

Comment:
Diffs seem confused here; this line is new. I do not expect that we want to force-enable interactive mode whenever `settings.interactive` is set; we may allow it to set defaults but this would force even agents with explicitly-disabled interactivity to be interactive. 

@pchalasani I don't see any existing use of `settings.interactive`, where do we use it? If we don't use it anywhere, we could change the type to `Optional[bool]` and only override if set.

Reply:
@nilspalumbo yes that is why I simplified that line, to not use `settings.interactive` -- there was a conflict I just resolved, it should no longer be here

Reply:
yes the diff-viewer in pycharm is getting confused

Reply:
Sounds good, thanks! I had to resort to running `diff` manually on blocks of code to see the changes